File: TEO/Mav.UI/Mav.Ui/src/app/models/search/user.search.model.ts
export class UserSearchModel {

  public static default = new UserSearchModel();

  public creatorID: string;

  public centreID?: number;

  public pageNo: number;

  public pageSize: number;

  public searchField: string;

  public searchTerm: string;

  public role?: string;

  public year?: number;

}

File: TEO/Mav.Models/SearchModels/UserSearchModel.cs
namespace Mav.Models.SearchModels
{
    public class UserSearchModel
    {
        public string CreatorID { get; set; }

        public int? CentreID { get; set; }

        public int PageNo { get; set; } = 1;

        public int PageSize { get; set; } = 10;

        public string SearchField { get; set; }

        public string SearchTerm { get; set; }

        public string Role { get; set; }

        public int? Year { get; set; }
    }
}
